src/CodeGen_LLVM.cpp

```diff
@@ -629,6 +629,19 @@ void CodeGen_LLVM::end_func(const std::vector<LoweredArgument>& args) {
     // Generate the function declaration and argument unpacking code.
     begin_func(f.linkage, simple_name, extern_name, f.args);
 
+    // If building with MSAN, ensure that calls to halide_msan_annotate_buffer_is_initialized()
+    // happen for every output buffer if the function succeeds.
+    if (f.linkage == LoweredFunc::External && target.has_feature(Target::MSAN)) {
```

Comment:
Might be cleaner to have a simple lowering pass that adds calls to annotate_buffer_is_initialized at the end of each produce (or the last update) node for each output. They'll be skipped automatically on error.